aspnetcore/release-notes/aspnetcore-5.0.md

```diff
@@ -0,0 +1,480 @@
+
+### Miscellaneous
+
+* The <xref:System.ComponentModel.DataAnnotations.CompareAttribute> can be applied to properties on Razor Page model.
+* Parameters and properties bound from the body are considered required by default. <!-- Review: How is this different from 3.1
+The validation system in .NET Core 3.0 and later treats non-nullable parameters or bound properties as if they had a [Required] attribute.
+see https://docs.microsoft.com/aspnet/core/mvc/models/validation?view=aspnetcore-3.1   
+-->
+
+## Web API
+
+### OpenAPI Specification on by default
+
+[OpenAPI Specification](http://spec.openapis.org/oas/v3.0.3) is a industry adopted convention for describing HTTP APIs and integrating them into complex business processes or with 3rd parties. OpenAPI is widely supported by all cloud providers and many API registries. Apps that emit OpenAPI documents from Web APIs have a variety of new opportunities in which those APIs can be used. In partnership with the maintainers of the open-source project [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore/), the ASP.NET Core API template contains a NuGet dependency on [Swashbuckle](https://github.com/domaindrivendev/Swashbuckle.AspNetCore), a popular open-source NuGet package that emits OpenAPI documents dynamically. Swashbuckle does this by introspecting over the API Controllers and generating the OpenAPI document at run-time, or at build time using the Swashbuckle CLI.
+
+In ASP.NET Core 5.0, the web API templates enable the OpenAPI support by default. To disable OpenAPI:
+
+* From the command line:
+
+	```dotnetcli
+	dotnet new webapi --no-openapi true`
+	```
+* From Visual Studio: Uncheck **Enable OpenAPI support**.
+
+All *.csproj* files created for web API projects contain the [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore/) NuGet package reference.
+
+```xml
+<ItemGroup>
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="5.5.1" />
+</ItemGroup>
+```
+
+The template generated code contains code in `Startup.ConfigureServices` that activates OpenAPI document generation:
+
+[!code-csharp[](~/release-notes/sample/StartupSwagger.cs?name=snippet)]
+
+The `Configure` method adds the Swashbuckle middleware, which enables the:
+
+* Document generation process.
+* Swagger UI page by default in development mode.
+
+The template generated code won't accidentally expose the API's description when publishing to production.
+
+[!code-csharp[](~/release-notes/sample/StartupSwagger.cs?name=snippet2)]
+
+#### Azure API Management Import
+
+When ASP.NET Core API projects enable OpenAPI, the Visual Studio 2019 version 16.8 Preview 3.2 and later publishing automatically offer an additional step in the publishing flow. Developers who use [Azure API Management](xref:tutorials/publish-to-azure-api-management-using-vs) have an opportunity to automatically import the APIs into Azure API Management during the publish flow:
```

Comment:
I don't think we need to refer to VS previews here. VS 168 will have shipped by the time .NET 5 ships.

```suggestion
When ASP.NET Core API projects enable OpenAPI, the Visual Studio 2019 version 16.8 and later publishing automatically offer an additional step in the publishing flow. Developers who use [Azure API Management](xref:tutorials/publish-to-azure-api-management-using-vs) have an opportunity to automatically import the APIs into Azure API Management during the publish flow:
```